src/kaitai_spec/attribute/enumeration.rs

```diff
@@ -0,0 +1,187 @@
+use crate::kaitai_spec::attribute::{Attribute, AttributeType, CommonKeys};
+use serde::de::Error;
+
+const ENDIAN: Endian = Endian::Big;
+#[derive(Debug, PartialEq)]
+pub struct Enumeration {
+    name: String,
+    type_: EnumType,
+}
+#[derive(Debug, PartialEq)]
+enum EnumType {
+    U1,
+    S1,
+    Long { type_: LongType, endian: Endian },
+}
+#[derive(Debug, PartialEq)]
+enum LongType {
+    U2,
+    U4,
+    U8,
+    S2,
+    S4,
+    S8,
+}
+#[derive(Debug, PartialEq, Clone)]
+enum Endian {
+    Little,
+    Big,
+}
+pub fn process<'de, A>(
+    name: String,
+    mut common_keys: CommonKeys,
+    mut map: A,
+) -> Result<Attribute, A::Error>
+where
+    A: serde::de::MapAccess<'de>,
+{
+    let mut endian: Endian = ENDIAN; //DEFAULT
+    let type_: Option<EnumType>;
+
+    while let Some(key) = map.next_key::<&str>()? {
+        if common_keys.process(key, &mut map)? {
+            continue;
+        }
+        match key {
+            _ => return Err(A::Error::unknown_field(key, &["nothing expected"])),
+        }
+    }
+    let type_unchecked: String = common_keys
+        .type_uncheck
+        .ok_or_else(|| A::Error::missing_field("type"))?;
+
+    match type_unchecked.chars().count() {
+        4 => {
+            let endian_unchecked: &str = &type_unchecked[2..4];
+            match endian_unchecked {
+                "le" => endian = Endian::Little,
+                "be" => endian = Endian::Big,
+                "" => {}
+                _ => return Err(A::Error::custom("invalid type")),
+            }
+        }
+        2 => {}
+        _ => return Err(A::Error::custom("invalid type")),
+    }
```

Comment:
```suggestion
    let endian = match type_unchecked.len() {
        4 => {
            match &type_unchecked[2..4] {
                "le" => Some(Endian::Little),
                "be" => Some(Endian::Big),
                _ => return Err(A::Error::custom("invalid type")),
            }
        }
        2 => None,
        _ => return Err(A::Error::custom("invalid type")),
    };
```

Reply:
I’ll remove the extra variable, but is it really necessary to change .chars().count() to .len()?

Reply:
Yep, what the concerns do you have?

Reply:
if refer to the official documentation:
![image](https://github.com/ArkadiyParovozow/ksrg/assets/145282159/49c14679-dc51-4525-8ecf-6a3386bc4de8)

Reply:
Possibly this could cause conflicts(?)

Reply:
Yep, it is correct, however we check the input against the string in ASCII encoding. So it's ok